generator/transit_generator.hpp

```diff
@@ -102,15 +104,74 @@ class DeserializerFromJson
   OsmIdToFeatureIdsMap const & m_osmIdToFeatureIds;
 };
 
-/// \brief Builds the transit section in the mwm.
+/// \brief The class contains all the information to make TRANSIT_FILE_TAG section.
+class GraphData
+{
+public:
+  void DeserializeFromJson(my::Json const & root, OsmIdToFeatureIdsMap const & mapping);
+  void SerializeToMwm(std::string const & mwmPath) const;
+  void AppendTo(GraphData const & rhs);
+  void Clear();
+  bool IsValid() const;
+
+  /// \brief Sorts all class fields execpt for |m_edges| by their ids.
+  void Sort();
+  /// \brief Calculates and updates |m_edges| by adding valid value for Edge::m_weight
+  /// if it's not valid.
+  /// \note |m_stops|, Edge::m_stop1Id and Edge::m_stop2Id in |m_edges| should be valid before call.
+  void CalculateEdgeWeights();
+  /// \brief Calculates best pedestrian segment for every gate in |m_gates|.
+  /// \note All gates in |m_gates| should have a valid |m_point| field before the call.
+  void CalculateBestPedestrianSegments(std::string const & mwmPath, std::string const & countryId);
+
+  std::vector<Stop> const & GetStops() const { return m_stops; }
+  std::vector<Gate> const & GetGates() const { return m_gates; }
+  std::vector<Edge> const & GetEdges() const { return m_edges; }
+  std::vector<Transfer> const & GetTransfers() const { return m_transfers; }
+  std::vector<Line> const & GetLines() const { return m_lines; }
+  std::vector<Shape> const & GetShapes() const { return m_shapes; }
+  std::vector<Network> const & GetNetworks() const { return m_networks; }
+
+private:
+  DECLARE_VISITOR_AND_DEBUG_PRINT(GraphData, visitor(m_stops, "stops"), visitor(m_gates, "gates"),
```

Comment:
Consider overriding clang-format here leaving one visitor per line.

Reply:
Yes it's more readable but less convenient to edit. (I'm using special plugin in my IDE for clang-format)...

Probably I'll do it later in a separate PR and for all transit visitors.